utils/googleutils/googleutils.go

```diff
@@ -3,12 +3,40 @@ package googleutils
 import (
 	"fmt"
 
+	"github.com/rudderlabs/rudder-server/config"
 	"golang.org/x/oauth2/google"
 )
 
+const (
+	EMPTY_CREDS   = "{}"
+	WI_CONFIG_KEY = "workloadIdentity"
+)
+
 func CompatibleGoogleCredentialsJSON(jsonKey []byte) error {
 	if _, err := google.ConfigFromJSON(jsonKey); err == nil {
 		return fmt.Errorf("Google Developers Console client_credentials.json file is not supported")
 	}
 	return nil
 }
+
+func ShouldSkipCredentialsInit(credentials string) bool {
+	return IsGKEEnabledWorkload() && IsCredentialsStringEmpty(credentials)
+}
+
+/*
+	The credentials are deemed to be empty when either the field credentials is
+	sent as empty string or when the field is set with "{}"
+
+	Note: This is true only for workload identity enabled rudderstack data-plane deployments
+*/
+func IsCredentialsStringEmpty(credentials string) bool {
+	return (credentials == "" || credentials == EMPTY_CREDS)
+}
+
+/*
+	We would check for rudder-server configuration for workload identity for google destinations
+*/
+func IsGKEEnabledWorkload() bool {
+	workloadType := config.GetString(fmt.Sprintf("%s.type", WI_CONFIG_KEY), "")
```

Comment:
If `workloadIdentity.type` has a value, is it possible to have any other value than `GKE`?

Reply:
It is possible to have another value called `EKS` but not sure about it as of this moment

Reply:
I was under the assumption that workload identity was a GKE thing only. If not and there are differences in workload identity configuration between GKE & EKS fine, otherwise a boolean config option would be more appropriate, e.g. `config.GetBool("WorkloadIdentityEnabled", false)`

Reply:
> If not and there are differences in workload identity configuration between GKE & EKS fine

Yes, there is a difference in workload identity configuration between GKE and EKS